src/Umbraco.Web.UI.Client/src/common/directives/components/property/umbproperty.directive.js

```diff
@@ -32,6 +35,19 @@ angular.module("umbraco.directives")
                 self.setPropertyError = function (errorMsg) {
                     $scope.property.propertyErrorMessage = errorMsg;
                 };
+                
+                var unsubscribe = $scope.$on("ExposePropertyEditorAPI", function(event, api) {
+                    
+                    //avoid eventual parent properties to capture this.
+                    event.stopPropagation();
+                    
+                    $scope.propertyEditorAPI = api;
```

Comment:
The name `propertyEditorApi` is too generic and doesn't really explain what this is doing. This is really completely specific to "Property Actions" (if that is the term we are going with)